lib/videoroom/meeting.ex

```diff
@@ -115,25 +165,43 @@ defmodule Videoroom.Meeting do
     raise "Connection to jellyfish closed!"
   end
 
-  defp handle_notification(%type{} = notification, state)
+  def handle_info({:peer_timeout, peer_id}, state) do
+    Logger.info("Peer #{peer_id} timed out")
+    delete_peer(peer_id, state)
+  end
+
+  defp handle_notification(%PeerConnected{peer_id: peer_id}, state) do
+    {timer, peer_timers} = Map.get_and_update(state.peer_timers, peer_id, fn p -> {p, nil} end)
+
+    Process.cancel_timer(timer)
+
+    {:noreply, %{state | peer_timers: peer_timers}}
+  end
+
+  defp handle_notification(%type{peer_id: peer_id}, state)
        when type in [PeerDisconnected, PeerCrashed] do
-    %{room_id: room_id, peer_id: peer_id} = notification
-    Room.delete_peer(state.client, room_id, peer_id)
-    {:ok, room} = Room.get(state.client, room_id)
```

Comment:
You used to rely on JF state, now you keep that locally. Don't you think that re-checking the room state via (`Room.get`) would be better

Reply:
I expect the Jellyfish to be empty, since upon creation the new peer is added to the local state and if a peer leaves the room, the local state is changed later. This means, that sometimes we can still have a peer in the local state but not in Jellyfish, but not the other way around.
I added error log in case this isn't true.